packages/dx-react-scheduler/src/plugins/confirmation-dialog.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import { getMessagesFormatter } from '@devexpress/dx-core';
+import {
+  Plugin,
+  Template,
+  TemplatePlaceholder,
+  TemplateConnector,
+  Action,
+} from '@devexpress/dx-react-core';
+import { ConfirmationDialogProps } from '../types/editing/confirmation-dialog.types';
+import { callActionIfExists, AppointmentModel } from '@devexpress/dx-scheduler-core';
+
+const defaultMessages = {
+  discardButton: 'Discard',
+  deleteButton: 'Delete',
+  cancelButton: 'Cancel',
+  confirmDeleteMessage: 'Are you sure you want to delete this appointment?',
+  confirmCancelMessage: 'Discard unsaved changes?',
+};
+
+const pluginDependencies = [
+  { name: 'EditingState' },
+];
+
+const ACTION_TYPES = {
+  CANCEL: 'cancel',
+  DELETE: 'delete',
+};
+
+const ConfirmationDialogBase: React.SFC<ConfirmationDialogProps> & {components: {
+  overlayComponent: string, containerComponent: string,
+  layoutComponent: string, buttonComponent: string,
+}} = ({
+  messages, overlayComponent: Overlay, layoutComponent: Layout, containerComponent: Container,
+  buttonComponent, ignoreDelete, ignoreCancel,
+}) => {
+  const getMessage = getMessagesFormatter({ ...defaultMessages, ...messages });
+  const modalContainer = React.useRef();
+
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [actionType, setActionType] = React.useState('');
+  const [hideActionName, setHideActionName] = React.useState('');
+  const [appointmentData, setAppointmentData] = React.useState({});
+
+  const toggleIsOpen = React.useCallback(() => {
+    setIsOpen(!isOpen);
+  }, [isOpen, setIsOpen]);
+
+  const confirmCancelChanges = React.useCallback((hideAction) => {
+    toggleIsOpen();
+    setHideActionName(hideAction);
+    setActionType(ACTION_TYPES.CANCEL);
+  }, [toggleIsOpen, setHideActionName, setActionType]);
+
+  const confirmDelete = React.useCallback(({
+    hideActionName: hideAction, appointmentData: changedAppointment,
+  }) => {
+    toggleIsOpen();
+    setHideActionName(hideAction);
+    setActionType(ACTION_TYPES.DELETE);
+    setAppointmentData(changedAppointment);
+  }, [toggleIsOpen, setHideActionName, setActionType, setAppointmentData]);
+
+  const confirmAction = React.useCallback((
+    isNewAppointment, hideEditor, stopEditAppointment, finishDeleteAppointment,
+    cancelAddedAppointment, cancelChangedAppointment,
+  ) => () => {
+    hideEditor();
+    toggleIsOpen();
+    if (isNewAppointment) {
+      callActionIfExists(cancelAddedAppointment, appointmentData);
+    } else {
+      callActionIfExists(stopEditAppointment, appointmentData);
+      callActionIfExists(cancelChangedAppointment, appointmentData);
+    }
+    if (actionType === ACTION_TYPES.DELETE && finishDeleteAppointment) {
+      finishDeleteAppointment(appointmentData);
+    }
```

Comment:
If we use `finishDeleteAppointment` we should add `IntegratedEditing` and `EditRecurrenceMenu` plugins as optional dependencies.